sdks/python/apache_beam/examples/kafkataxi/kafka_taxi.py

```diff
@@ -43,34 +52,43 @@ def run(bootstrap_servers, topic, with_metadata, pipeline_args):
   #                  '--num_workers', 'my-num-workers',
   #                  '--experiments', 'use_runner_v2']
 
-  pipeline_options = PipelineOptions(
-      pipeline_args, save_main_session=True, streaming=True)
   window_size = 15  # size of the Window in seconds.
 
-  def log_ride_with_metadata(record):
+  def log_ride(ride):
+    if 'timestamp' in ride:
+      logging.info(
+          'Found ride at latitude %r and longitude %r with %r '
+          'passengers at timestamp %r',
+          ride['latitude'],
+          ride['longitude'],
+          ride['passenger_count'],
+          ride['timestamp'])
+    else:
+      logging.info(
+          'Found ride at latitude %r and longitude %r with %r '
+          'passengers',
+          ride['latitude'],
+          ride['longitude'],
+          ride['passenger_count'])
+
+  def convert_kafka_record_to_dictionary(record):
+    if hasattr(record, 'value'):
```

Comment:
Let's add a comment here to clarify. ("if" condition becomes true when reading with metadata).